oat-sa/tao-qunit-testrunner-fe
# Feature/ADF-1417/Upgrade to Node 18

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1419](https://oat-sa.atlassian.net/browse/ADF-1419)

### Summary

Set Node 18 for the CI workflow.

Update the devDependencies accordingly.

BREAKING CHANGE:
- `minimatch` has been upgraded from `3.0.4` to `9.0.0`.
- `glob` has been upgrade from `7.1.6` to `8.1.0`.
- `\` is no longer supported as a path separator
- require Node `14.17+`

### Details

Added feature: the timeout is displayed together with the related error message.

To properly support Node 18, the dev dependencies had to be updated and aligned.

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

It also removes vulnerabilities 🎉 
```
audited 536 packages in 2.01s

found 0 vulnerabilities
```

### How to test
- check the report on GitHub
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package: `npm i`
- run the unit tests: `npm test`, `npm run test:cov`, `npm run coverage`
- test the linter: `npm run lint`

[ADF-1417]: https://oat-sa.atlassian.net/browse/ADF-1417?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ADF-1419]: https://oat-sa.atlassian.net/browse/ADF-1419?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
